src/test/java/lambda/part3/exercise/Exercise2.java

```diff
@@ -0,0 +1,162 @@
+         * Для добавления в новый список каждый элемент преобразовывается с использованием заданной функции.
+         * ([T], (T -> R)) -> [R]
+         * @param mapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> map(Function<T, R> mapping) {
+            List<R> list = new ArrayList<>();
+            for (T t : source) {
+                list.add(mapping.apply(t));
+            }
+            return from(list);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается в список с использованием заданной функции.
+         * ([T], (T -> [R])) -> [R]
+         * @param flatMapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
+            List<R> list = new ArrayList<>();
+            for (T t : source) {
+                list.addAll(flatMapping.apply(t));
+            }
+            return new MapHelper<>(list);
+        }
+    }
+
+    @Test
+    public void mapEmployeesToLengthOfTheirFullNamesUsingMapHelper() {
+        List<Employee> employees = getEmployees();
+
+        // TODO                 MapHelper.from(employees)
+        // TODO                          .map(Employee -> Person)
+        // TODO                          .map(Person -> String(full name))
+        // TODO                          .map(String -> Integer(length of string))
+        // TODO                          .getMapped();
+        List<Integer> lengths = MapHelper.from(employees)
+                .map(Employee::getPerson)
+                .map(Person::getFullName)
+                .map(String::length)
+                .getMapped();
+
+        assertEquals(Arrays.asList(14, 19, 14, 15, 14, 16), lengths);
+    }
+
+    @Test
+    public void mapEmployeesToCodesOfLetterTheirPositionsUsingMapHelper() {
+        List<Employee> employees = getEmployees();
+
+        // TODO               MapHelper.from(employees)
+        // TODO                        .flatMap(Employee -> JobHistoryEntry)
+        // TODO                        .map(JobHistoryEntry -> String(position))
+        // TODO                        .flatMap(String -> Character(letter))
+        // TODO                        .map(Character -> Integer(code letter)
+        // TODO                        .getMapped();
+        List<Integer> codes = MapHelper.from(employees)
+                .flatMap(Employee::getJobHistory)
+                .map(JobHistoryEntry::getPosition)
+                .flatMap((String string) -> {
+                    char[] charArr = string.toCharArray();
+                    List<Character> list = new ArrayList<>();
+                    for (char ch : charArr) {
+                        list.add(ch);
+                    }
+                    return list;
+                })
```

Comment:
It could be better if you extract this lambda to new method and place here reference to it.